modal/_utils/grpc_utils.py

```diff
@@ -208,6 +215,9 @@ async def retry_transient_errors(
 
             n_retries += 1
 
+            if retry_warning_message and n_retries % retry_warning_message.warning_interval == 0:
+                logger.warning(retry_warning_message.message)
+
```

Comment:
This is the heart of the change - rather than letting the exception propagate to the caller, and logging the warning there - we now log the warning here.